### PR TITLE
Simparse

### PR DESCRIPTION
**Pull request by Holado**
Company (fyrirtæki) tokens added, Tokenizer and Reynir have been updated to handle them.

### CODE REVIEW

**Comment by Holado** [comment]
`_CORPORATION_ENDINGS` was moved from binparser.py to bintokenizer.py, I haven't yet deleted it from binparser.py but it isn't used there anymore.

File: src/reynir/bintokenizer.py [hunk 1]
```diff
@@ -289,6 +289,49 @@ def all_except_suffix(s):
 # Given names that can also be family names (and thus gender- and caseless as such)
 BOTH_GIVEN_AND_FAMILY_NAMES = frozenset(("Hafstein",))
 
+# Note: these must have a meaning for this to work, so specifying them
+# as abbreviations to Main.conf is recommended
+_CORPORATION_ENDINGS = frozenset(
```
Comment (jokull) [inline src/reynir/bintokenizer.py:294]: Maybe add "hses." for [húsnæðissjálfseignastofnun](https://www.ils.is/stofnframlog/husnaedissjalfseignastofnun/).

File: src/reynir/bintokenizer.py [hunk 2]
```diff
@@ -1135,6 +1177,45 @@ def parse_phrases_3(token_stream, token_ctor):
...
+                while True:
+                    if next_token.txt in _CORPORATION_ENDINGS:
+                        # Form Company-token, stop searching
+                        entitytxt = entitytxt + " " + next_token.txt
+                        token = token_ctor.Company(entitytxt)
+                        next_token = next(token_stream)
+                        break
+                    elif (
+                        (next_token.kind == TOK.ENTITY or (next_token.kind == TOK.WORD and not token.val))
+                        and next_token.txt[0].isupper() and next_token.txt[1:].islower()
```
Comment (vthorsteinsson) [inline src/reynir/bintokenizer.py:1204]: Better: `next_token.txt.istitle()`

File: src/reynir/bintokenizer.py [hunk 3]
```diff
@@ -1135,6 +1177,45 @@ def parse_phrases_3(token_stream, token_ctor):
...
+                    ):
+                        entitytxt = entitytxt + " " + next_token.txt
```
Comment (vthorsteinsson) [inline src/reynir/bintokenizer.py:1206]: Better: `entitytxt += " " + next_token.txt`

File: src/reynir/bintokenizer.py [hunk 4]
```diff
@@ -1135,6 +1177,45 @@ def parse_phrases_3(token_stream, token_ctor):
...
+                # TODO allow more than one name to be merged?
+                entitytxt = token.txt
+                found = False
+                while True:
+                    if next_token.txt in _CORPORATION_ENDINGS:
+                        # Form Company-token, stop searching
+                        entitytxt = entitytxt + " " + next_token.txt
+                        token = token_ctor.Company(entitytxt)
+                        next_token = next(token_stream)
+                        break
```
Comment (vthorsteinsson) [inline src/reynir/bintokenizer.py:1201]: Add: `found = False`

Comment (Holado) [reply src/reynir/bintokenizer.py:1204]: Done!

File: src/reynir/bintokenizer.py (hunk 3 above)
Comment (Holado) [reply src/reynir/bintokenizer.py]: Done!

File: src/reynir/bintokenizer.py (hunk 2 above)
Comment (Holado) [reply src/reynir/bintokenizer.py]: Done!

File: src/reynir/bintokenizer.py [hunk 5]
```diff
@@ -1135,6 +1177,45 @@ def parse_phrases_3(token_stream, token_ctor):
...
+                # Upper-case word: Check next word
+                # Most likely two unknown person names
+                # Can also be a corporation ending
+                # TODO allow more than one name to be merged?
+                entitytxt = token.txt
+                found = False
+                while True:
+                    if next_token.txt in _CORPORATION_ENDINGS:
+                        # Form Company-token, stop searching
+                        entitytxt = entitytxt + " " + next_token.txt
```
[the remark this answers is not in the document]
Comment (Holado) [reply src/reynir/bintokenizer.py]: Done!

File: src/reynir/bintokenizer.py [hunk 6]
```diff
@@ -1135,6 +1177,45 @@ def parse_phrases_3(token_stream, token_ctor):
                     ]
                 )
                 next_token = next(token_stream)
+            elif (
+                (
+                    token.kind == TOK.ENTITY 
+                    or (token.kind == TOK.WORD and not token.val)
+                    or (token.kind == TOK.WORD and token.val[0].ordfl == "entity")
+                )
+                and token.txt[0].isupper() and token.txt[1:].islower()
```
[the remark this answers is not in the document]
Comment (Holado) [reply src/reynir/bintokenizer.py]: Done!

File: src/reynir/bintokenizer.py (hunk 1 above)
Comment (Holado) [reply src/reynir/bintokenizer.py:294]: Done!

**Pull request merged**